cli-plugins/plugin/plugin.go

```diff
@@ -46,6 +46,22 @@ func RunPlugin(dockerCli *command.DockerCli, plugin *cobra.Command, meta manager
 				opts = append(opts, withPluginClientConn(plugin.Name()))
 			}
 			err = tcmd.Initialize(opts...)
+			ogRunE := cmd.RunE
+			if ogRunE == nil {
+				ogRun := cmd.Run
+				//nolint:unparam // necessary because error will always be nil here
+				ogRunE = func(cmd *cobra.Command, args []string) error {
+					ogRun(cmd, args)
+					return nil
+				}
+				cmd.Run = nil
+			}
+			cmd.RunE = func(cmd *cobra.Command, args []string) error {
+				stopInstrumentation := dockerCli.StartInstrumentation(ctx, cmd)
+				err := ogRunE(cmd, args)
+				stopInstrumentation(err)
+				return err
+			}
```

Comment:
Hmm, do we actually need to wrap the `cmd.RunE` here?

Wouldn't it be enough to call the Start/stopInstrumentation around the `cmd.Execute` call?
https://github.com/docker/cli/blob/8f0b236c2848bfbda9e65240a1b97c2f009a2caf/cli-plugins/plugin/plugin.go#L76

Reply:
Noo, needs to be there, due to when it gets called. The plugins do necessary setup in their `PersistentPreRunE`, this has to happen after that.